### PR TITLE
Boyer Moore Algorithm in C++

### PR DESCRIPTION
**Pull request by Avantika782**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Pattern Searching using Boyer Morey Algorithm in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue Number : #3838

### CODE REVIEW

**Comment by Avantika782** [comment]
Please review this;)
@SKAUL05 @nrickles3 @ArchanGhosh @hushee69

**Comment by Avantika782** [comment]
@siddharth25pandey @Siddhant-K-code Please review it.

File: C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/* C++ Program for Bad Character Heuristic of Boyer 
+Moore String Matching Algorithm */
+#include <bits/stdc++.h>
+using namespace std;
+#define NO_OF_CHARS 256
+
+// The preprocessing function for Boyer Moore's
+// bad character heuristic
+void badCharHeuristic(string str, int size,
+                      int badchar[NO_OF_CHARS])
```
Comment (Siddhant-K-code) [inline C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp]: Format This Correctly!

File: C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    string txt, pat;
+    cout << "Enter the string: ";
+    cin >> txt;
+    cout << endl;
+    cout << "Enter the pattern to search in string: ";
+    cin >> pat;
+    cout << endl;
+    search(txt, pat);
+    return 0;
+}
```
Comment (Siddhant-K-code) [inline C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp:90]: Add Time Complexity and Example Test Cases

**Review by Siddhant-K-code — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp (hunk 2 above)
Comment (Avantika782) [reply C-Plus-Plus/cp/Boyer_Moore_Algorithm.cpp:90]: Added 
Please review it. @Siddhant-K-code

**Review by Siddhant-K-code — CHANGES_REQUESTED** [review]
Try to Resolve merge conflict!!

**Comment by Avantika782** [comment]
Resolved merge conflicts.

**Comment by Avantika782** [comment]
Resolved merge conflicts. @Siddhant-K-code Review it once again ;)

**Review by Siddhant-K-code — APPROVED** [review]
LGTM!

**Review by HarshCasper — APPROVED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
code already present #1730

**Pull request merged**